src/object/srv_obj_migrate.c

```diff
@@ -3853,49 +3934,302 @@ ds_obj_migrate_handler(crt_rpc_t *rpc)
+
+	d_iov_set(&key_iov, &uoid, sizeof(uoid));
+	d_iov_set(&val_iov, &val, sizeof(struct migrate_obj_val));
+	return dbtree_lookup(cont_toh, &key_iov, &val_iov);
+}
+
+#define REINT_ITER_YIELD_CNT	(256)
+struct reint_post_iter_arg {
+	daos_handle_t			 ria_migrated_tree_hdl;
+	struct migrate_pool_tls		*ria_tls;
+	daos_handle_t			 ria_cont_toh;
+	uuid_t				 ria_co_uuid;
+	int				 ria_yield_cnt;
+};
+
+static int
+reint_post_obj_iter_cb(daos_handle_t ch, vos_iter_entry_t *entry,
+		       vos_iter_type_t type, vos_iter_param_t *iter_param,
+		       void *data, unsigned *acts)
+{
+	struct reint_post_iter_arg	*arg = data;
+	struct migrate_pool_tls		*tls = arg->ria_tls;
+	daos_unit_oid_t			 uoid = entry->ie_oid;
+	int				 rc = 0;
+
+	/* for non-replica obj don't try to delete it */
+	if (daos_obj_id2ord(uoid.id_pub) == OR_RP_1)
+		goto out;
+
+	rc = obj_tree_lookup_uoid(arg->ria_cont_toh, uoid);
+	if (rc) {
+		if (rc == -DER_NONEXIST) {
+			D_DEBUG(DB_TRACE, DF_RB": cont "DF_UUID", uoid "DF_UOID
+				" non-exist in migrate tree, discard it.\n",
+				DP_RB_MPT(tls), DP_UUID(arg->ria_co_uuid), DP_UOID(uoid));
+			rc = vos_obj_delete(iter_param->ip_hdl, uoid);
+			if (rc) {
+				DL_ERROR(rc, DF_RB " vos_obj_delete, cont "DF_UUID", obj "DF_UOID
+					 " failed", DP_RB_MPT(tls), DP_UUID(arg->ria_co_uuid),
+					 DP_UOID(uoid));
+				goto out;
+			}
+			*acts |= VOS_ITER_CB_DELETE;
+			goto out;
+		}
+
+		DL_ERROR(rc, DF_RB " obj_tree_lookup_uoid "DF_UOID" failed",
+			 DP_RB_MPT(tls), DP_UOID(uoid));
+		goto out;
+	}
+
+out:
+	if (--arg->ria_yield_cnt <= 0) {
+		D_DEBUG(DB_REBUILD, DF_RB " rebuild yield: %d\n", DP_RB_MPT(tls), rc);
+		arg->ria_yield_cnt = REINT_ITER_YIELD_CNT;
+		if (rc == 0)
+			dss_sleep(0);
+		*acts |= VOS_ITER_CB_YIELD;
+	}
+	return rc;
+}
+
+static int
+reint_post_cont_iter_cb(daos_handle_t ih, vos_iter_entry_t *entry,
+			vos_iter_type_t type, vos_iter_param_t *iter_param,
+			void *data, unsigned *acts)
+{
+	struct reint_post_iter_arg	*arg = data;
+	struct migrate_pool_tls		*tls = arg->ria_tls;
+	vos_iter_param_t		 param = { 0 };
+	struct vos_iter_anchors		 anchor = { 0 };
+	daos_handle_t			 toh = arg->ria_migrated_tree_hdl;
+	daos_handle_t			 coh;
+	int				 rc;
+
+	rc = obj_tree_lookup_cont(toh, entry->ie_couuid, &arg->ria_cont_toh);
+	if (rc) {
+		if (rc == -DER_NONEXIST) {
+			D_DEBUG(DB_TRACE, DF_RB": cont "DF_UUID" non-exist in migrate tree, "
+				"discard it.\n", DP_RB_MPT(tls), DP_UUID(entry->ie_couuid));
+			rc = ds_cont_child_destroy(tls->mpt_pool_uuid, entry->ie_couuid);
+			if (rc) {
+				DL_ERROR(rc, DF_RB " destroy container "DF_UUID" failed",
+					 DP_RB_MPT(tls), DP_UUID(entry->ie_couuid));
+				goto out;
+			}
+			*acts |= VOS_ITER_CB_DELETE;
+			goto out;
+		}
+
+		DL_ERROR(rc, DF_RB " obj_tree_lookup_cont "DF_UUID" failed",
+			 DP_RB_MPT(tls), DP_UUID(entry->ie_couuid));
+		goto out;
+	}
+
+	D_ASSERT(daos_handle_is_valid(arg->ria_cont_toh));
+
+	rc = vos_cont_open(iter_param->ip_hdl, entry->ie_couuid, &coh);
```

Comment:
Here we intentionally call VOS interface to open the container? I think the ds_cont_child in container level should be in started state? Why don't we just lookup & use  ds_cont_child directly?

Reply:
I'll see if the ds_cont_child can be used, thx